iron-inttest/src/test/java/io/axway/iron/spi/kafka/Utils.java

```diff
@@ -0,0 +1,141 @@
+package io.axway.iron.spi.kafka;
+
+import java.io.*;
+import java.net.DatagramSocket;
+import java.net.ServerSocket;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.*;
+import java.util.concurrent.*;
+import java.util.concurrent.atomic.*;
+import com.google.common.util.concurrent.UncheckedExecutionException;
+
+import static io.axway.alf.assertion.Assertion.checkState;
+import static java.lang.Thread.currentThread;
+import static java.util.Comparator.*;
+import static java.util.stream.Collectors.*;
+
+/**
+ * This class contains utility methods for the Kafka SPI tests
+ */
+final class Utils {
+    private static final AtomicInteger PORT = new AtomicInteger(1024);
+
+    /**
+     * Provides an available port
+     *
+     * @return available port
+     */
+    static int providePort() {
+        while (true) {
+            // Pick a port
+            int port = PORT.incrementAndGet();
+            checkState(port <= 0xFFFF, "No more ports left to use in test");
+
+            // Check if the port is not already used by another process
+            try (ServerSocket ss = new ServerSocket(port); DatagramSocket ds = new DatagramSocket(port)) {
+                ss.setReuseAddress(true);
+                ds.setReuseAddress(true);
+                return port;
+            } catch (IOException ignored) {
+                // continue, try again
+            }
+        }
+    }
+
+    /**
+     * Tries to recursively delete a folder, silently give up if it cannot be deleted.
```

Comment:
it doesn't seem to be 'silent' since a UncheckedIOException is thrown in case of problem

Reply:
The deletion failure is silent but if you cannot start walking the file structure then it throws the unchecked exception